File: solution/2610. Convert an Array Into a 2D Array With Conditions/2610_02.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        const int n = nums.size();
        vector<int> count(n + 1); // nums[i] <= nums.length
        
        vector<vector<int>> res;
        for (int x : nums) {
            if (count[x] >= res.size())
                res.push_back({});
            res[count[x]++].push_back(x);
        }
        
        return res;
    }
};

File: solution/2610. Convert an Array Into a 2D Array With Conditions/2610_02.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        d = defaultdict(lambda: 0)
        res = []
        for x in nums:
            if d[x] >= len(res):
                res.append([])
            res[d[x]].append(x)
            d[x] += 1
        return res
    
